lib/ruby_lsp/requests/document_link.rb

```diff
@@ -109,14 +109,14 @@ def visit_comment(node)
       # 1. The version in the URI
       # 2. The version in the RBI file name
       # 3. The version from the gemspec
-      sig { params(uri: URI::Source).returns(T.nilable(String)) }
+      sig { params(uri: URI::Source).returns(String) }
       def resolve_version(uri)
         version = uri.gem_version
         return version unless version.nil? || version.empty?
 
         return @gem_version unless @gem_version.nil? || @gem_version.empty?
 
-        GEM_TO_VERSION_MAP[uri.gem_name]
+        GEM_TO_VERSION_MAP.fetch(uri.gem_name)
```

Comment:
I think the mistake was actually the other way around. If the source comment is for a gem that is no longer a part of the bundle, then `GEM_TO_VERSION_MAP[uri.gem_name]` will return `nil`.

Instead of doing the `T.must`, we need to return early in that case.

Reply:
Yeah I agree with Vini here. `nil` in this case is a valid output. We should do an early return on line 95 instead.

Reply:
Ok, I've put it back to how it was previously and created a separate issue:

https://github.com/Shopify/ruby-lsp/issues/464